src/components/App.js

```diff
@@ -8,6 +8,29 @@ import { Totals } from './Totals.js';
 import { Menu } from './Menu.js';
 import { Lazy } from './Lazy.js';
 import { router } from '../router.js';
+import { dialogService } from '../services/dialogService.js';
+import { debounce } from 'lodash-es';
+
+const debouncedCloseDialogService = debounce(
+  (params) => {
+    let status = {
+      menuDialogClosed: false,
+      countryDialogClosed: false,
+    };
+
+    if (params.has('country') || params.has('iso2')) {
```

Comment:
should the params be looked up dynamically, otherwise you may be looking at old params by the time this function runs? pretty extremely, but since we're taking the effort of debouncing this.

Reply:
Not sure I follow, the params are passed in `__closeDialog`, the first time the function is called, and the debounce is leading, so it should only take those params

Reply:
Ah ok, yeah then it will be in sync. Though conceptually I think it makes more sense to look them up when you need them, then they are more aligned in the code.

Reply:
Ahh yes, I see what you mean now. Makes sense, I'll change that